src/dist/component/package.rs

```diff
@@ -182,9 +182,23 @@ fn unpack_ram(
         .ok()
         .and_then(|budget_str| budget_str.parse::<usize>().ok())
     {
-        // Note: In future we may want to add a warning or even an override if a user
-        // supplied budget is larger than effective_max_ram.
-        Some(budget) => budget,
+        Some(budget) => {
+            if budget < minimum_ram {
+                warn!(
+                    "Specified unpack RAM limit ({}) less than minimum of {}.  Unpacking may fail.",
+                    budget, minimum_ram
+                );
+            }
+            if budget > default_max_unpack_ram {
+                warn!(
+                    "Specified unpack RAM limit ({}) greater than detected available RAM of {}.  Clamping to detected limit.",
+                    budget, default_max_unpack_ram
+                );
+                default_max_unpack_ram
+            } else {
+                budget
+            }
+        }
```

Comment:
I like the clamping to max ram.
I have a suggested different message - something along these lines, which is what I drafted before the meeting I was in delayed me...
```warn!(
+                    "Ignoring RUSTUP_UNPACK_RAM ({}) less than {}.",
+                    budget, minimum_ram
+                );```